### DISCUSSION TITLE
Integrating DIDs with methods apart from the current provided ones

### DISCUSSION THREAD

**Discussion by atharva-pb** [walt-id/.github#64] [Help] [answered]
Hi Walt team,

I am interested in issuing, importing and storing a DID with my own custom method or a third-party method in Walt ID wallet. Is there a way to issue and store DIDs with methods other than the currently supported ones (key, web, etc.)? We have explored the APIs of SSIKIT (Signatory, Custodian, Auditor), and they are working fine with the given methods, but I want to experiment with a custom DID method. What approach should I take to solve this problem?

Thank you!

**Comment by philpotisk** [comment]
Hi @atharva-pb thanks for your question! 

The SSI Kit can indeed be extended by any DID method. Please take a look at how the existing methods are implemented. https://github.com/walt-id/waltid-ssikit/tree/main/src/main/kotlin/id/walt/services/did/ basically you would need to implement the functionality to register and to resolve the DID document. 

Which DID do you want to use? We are happy to support you with this task.

**Reply by atharva-pb** [reply]
Hi @philpotisk,

Thank you for your response. I am interested in using the "polygonid" method whose specifications can be found [here](https://github.com/0xPolygonID/did-polygonid/blob/8695f70d060c5884137fe18c549dfae07b5a7053/did-polygonid-method-draft.md#L4). I would be grateful for your support in implementing this method with the SSI Kit. Could you please advise me on the steps I should take to achieve this?

Best,
AB

**Reply by philpotisk** [reply, marked as answer]
I can think of two options how to implement DID Polygon:
1.) As I found this [REST API](https://0xpolygonid.github.io/tutorials/issuer-node/issuer-node-api/identity/apis/) that seems to support the required functionality, you could utilize these endpoints, similar as we integrated [did:cheqd for did-resolution](https://github.com/walt-id/waltid-ssikit/blob/main/src/main/kotlin/id/walt/services/did/resolvers/DidCheqdResolver.kt) and for [did-registration](https://github.com/walt-id/waltid-ssikit/blob/main/src/main/kotlin/id/walt/services/ecosystems/cheqd/CheqdService.kt)
2.) You could implement the Polygon ID spec natively, similar as we implemented DID EBSI ([resolve](https://github.com/walt-id/waltid-ssikit/blob/main/src/main/kotlin/id/walt/services/did/resolvers/DidEbsiResolver.kt), [register](https://github.com/walt-id/waltid-ssikit/blob/main/src/main/kotlin/id/walt/services/ecosystems/essif/didebsi/WaltIdDidEbsiService.kt))
